src/Symfony/Component/Security/Core/User/UserChecker.php

```diff
@@ -28,10 +28,14 @@ class UserChecker implements UserCheckerInterface
      */
     public function checkPreAuth(UserInterface $user)
     {
-        if (!$user instanceof AdvancedUserInterface) {
+        if (!$user instanceof AdvancedUserInterface && !$user instanceof User) {
```

Comment:
Previously it would return early if it wasn't an AdvancedUserInterface. In the future, this will be only if it's not a User (InMemoryUser). That's what I meant with the confusion about the naming.